tests/test_dijkstra.py

```diff
@@ -39,6 +39,240 @@ def test_dijkstra(self):
         expected = {4: 23.0}
         self.assertEqual(expected, path)
 
+    def test_dijkstra_path(self):
+        g = retworkx.PyDiGraph()
+        a = g.add_node("A")
+        b = g.add_node("B")
+        c = g.add_node("C")
+        d = g.add_node("D")
+        e = g.add_node("E")
+        f = g.add_node("F")
+        edge_list = [
+            (a, b, 7),
+            (c, a, 9),
+            (a, d, 14),
+            (b, c, 10),
+            (d, c, 2),
+            (d, e, 9),
+            (b, f, 15),
+            (c, f, 11),
+            (e, f, 6),
+        ]
+        g.add_edges_from(edge_list)
+        paths = retworkx.digraph_dijkstra_shortest_paths(g, a)
+        expected = {
+            1: [0, 1],
+            2: [0, 3, 2],
+            3: [0, 3],
+            4: [0, 3, 4],
+            5: [0, 1, 5],
+        }
+        self.assertEqual(expected, paths)
+
+    def test_dijkstra_path_with_weight_fn(self):
+        g = retworkx.PyDiGraph()
+        a = g.add_node("A")
+        b = g.add_node("B")
+        c = g.add_node("C")
+        d = g.add_node("D")
+        e = g.add_node("E")
+        f = g.add_node("F")
+        edge_list = [
+            (a, b, 7),
+            (c, a, 9),
+            (a, d, 14),
+            (b, c, 10),
+            (d, c, 2),
+            (d, e, 9),
+            (b, f, 15),
+            (c, f, 11),
+            (e, f, 6),
+        ]
+        g.add_edges_from(edge_list)
+        paths = retworkx.digraph_dijkstra_shortest_paths(
+            g, a, weight_fn=lambda x: x)
+        expected = {
+            1: [0, 1],
+            2: [0, 1, 2],
```

Comment:
The other one doesn't find this path because of the ordering of the edges, right? And here this one is found because it takes the weights into account?

Reply:
Yeah, the weight value is set to an default of `1.0` in the above test method, so the path it returns is based on the binary heap insertion order (LIFO iirc). While here the weight is being taken into account so a-> b has a weight of 7 vs a->e having a weight of 14.